R/add_pvalue.R

```diff
@@ -0,0 +1,111 @@
+#' Add p-value
+#'
+#' @description
+#' - `add_pvalue("caption")`: Add a p-value to the figure via `ggplot2::labs(caption=)`
+#' - `add_pvalue("annotation")`: Add a p-value text annotation via `ggplot2::annotation("text")`
+#'
+#' P-values are calculated with `survival::survdiff()`.
+#' Examples of custom placement located in the help file for `survfit_p()`.
+#'
+#' @param location string indicating where to place p-value. Must be one of
+#' `c("caption", "annotation")`
+#' @param caption string to be placed as the caption/annotation. String will
+#' be processed with `glue::glue()`, and the default is "{p.value}"
+#' @inheritParams survfit2_p
+#' @inheritParams survival::survdiff
+#' @param ... arguments passed to `ggplot2::annotate()`
+#'
+#' @return a ggplot2 figure
+#' @export
+#'
+#' @seealso `survfit_p()`
+#' @examples
+#' survfit2(Surv(time, status) ~ surg, df_colon) %>%
+#'   ggsurvfit() +
+#'   add_pvalue()
+#'
+#' survfit2(Surv(time, status) ~ surg, df_colon) %>%
+#'   ggsurvfit() +
+#'   add_pvalue("annotation", size = 5)
```

Comment:
consider an additional example to illustrate passing in additional arguments. e.g. 

```
# Passing in arguements to control "annotation" positioning 
survfit2(Surv(time, status) ~ surg, df_colon) %>%
  ggsurvfit() + 
  add_pvalue(location = "annotation", size = 5, x = 1.0, y = 0.4)
```

Reply:
nice! i added more details about using these args to place the p-value